File: python/TDA/nodo/nodo.py
class Nodo:
    def __init__(self, vertice):
        self.vertice = vertice # Vertice
        self.adyacentes = {}  # Diccionario de vecinos y sus pesos
        self.distancia = float('inf')  # Inicia con distancia infinita
        self.predecesor = None  # Nodo previo en el camino más corto

    def configurarAdyacente(self, adyacente, peso):
        self.adyacentes[adyacente] = peso

    def __repr__(self):
        # Obtener datos sobre el nodo
        return f"Nodo({self.vertice}, distancia={self.distancia}, predecesor={self.predecesor})"

File: cpp/TDA/nodo/Nodo.cpp
#include "Nodo.h"
#include <tuple>
#include <limits> 

Nodo::Nodo(std::tuple<int, int> coordenadas)
{
    this->coordenadas = coordenadas;
    distancia = std::numeric_limits<float>::infinity();  // distancia infinita
    adyacentes = {};
    predecesor = nullptr;
}

// Getters
float Nodo::getDistancia() const
{
    return distancia;
}

Nodo *Nodo::getPredecesor() const
{
    return predecesor;
}
std::tuple<int,int> Nodo::getCoordenadas() const
{
    return coordenadas;
}
std::map<std::tuple<int,int>, int> Nodo::getAdyacentes() const
{
    return adyacentes;
}
// Setters
void Nodo::setDistancia(float distancia)
{
    this->distancia = distancia;
}

void Nodo::setPredecesor(Nodo* predecesor)
{
    this->predecesor = predecesor;
}
void Nodo::setAdyacentes(std::map<std::tuple<int,int>, int> adyacentes)
{
    this->adyacentes = adyacentes;
}
void Nodo::setCoordenadas(std::tuple<int, int> coordenadas)
{
    this->coordenadas = coordenadas;
}

void Nodo::configurarAdyacente(std::tuple<int, int> coordenadas, float peso) {
    this->adyacentes[coordenadas] = peso;  
}